### ISSUE TITLE
Certain strings used as function names have odd behavior when the function argument is a Python float.

### ISSUE THREAD

**Issue by moorepants** [sympy/sympy#12716]
I'm not sure why these functions evaluate to floating point values. Shouldn't they remain as `psi(0.0)` and `phi(0.0)`.

```
Python 3.5.3 |Continuum Analytics, Inc.| (default, Mar  6 2017, 11:58:13) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.1.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import sympy as sm

In [2]: t = sm.symbols('t')

In [3]: phi, psi = sm.symbols('phi, psi', cls=sm.Function)

In [4]: phi(t)
Out[4]: phi(t)

In [5]: psi(t)
Out[5]: psi(t)

In [6]: phi(0)
Out[6]: phi(0)

In [7]: phi(0.0)
Out[7]: 1.61803398874989

In [8]: psi(0)
Out[8]: psi(0)

In [9]: psi(0.0)
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/miniconda3/lib/python3.5/site-packages/sympy/core/evalf.py in evalf(x, prec, options)
   1276     try:
-> 1277         rf = evalf_table[x.func]
   1278         r = rf(x, prec, options)

KeyError: psi

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-9-02da59d89871> in <module>()
----> 1 psi(0.0)

~/miniconda3/lib/python3.5/site-packages/sympy/core/function.py in __new__(cls, *args, **options)
    718     def __new__(cls, *args, **options):
    719         args = list(map(sympify, args))
--> 720         obj = super(AppliedUndef, cls).__new__(cls, *args, **options)
    721         return obj
    722 

~/miniconda3/lib/python3.5/site-packages/sympy/core/function.py in __new__(cls, *args, **options)
    390         pr2 = min(cls._should_evalf(a) for a in result.args)
    391         if pr2 > 0:
--> 392             return result.evalf(mlib.libmpf.prec_to_dps(pr))
    393         return result
    394 

~/miniconda3/lib/python3.5/site-packages/sympy/core/evalf.py in evalf(self, n, subs, maxn, chop, strict, quad, verbose)
   1383             options['quad'] = quad
   1384         try:
-> 1385             result = evalf(self, prec + 4, options)
   1386         except NotImplementedError:
   1387             # Fall back to the ordinary evalf

~/miniconda3/lib/python3.5/site-packages/sympy/core/evalf.py in evalf(x, prec, options)
   1282             if 'subs' in options:
   1283                 x = x.subs(evalf_subs(prec, options['subs']))
-> 1284             re, im = x._eval_evalf(prec).as_real_imag()
   1285             if re.has(re_) or im.has(im_):
   1286                 raise NotImplementedError

~/miniconda3/lib/python3.5/site-packages/sympy/core/function.py in _eval_evalf(self, prec)
    501 
    502         with mpmath.workprec(prec):
--> 503             v = func(*args)
    504 
    505         return Expr._from_mpmath(v, prec)

TypeError: psi() missing 1 required positional argument: 'z'
```

**Comment by adityausathe** [comment]
> I'm not sure why these functions evaluate to floating point values. Shouldn't they remain as psi(0.0) and phi(0.0).


- Any `f(<float>)` call first checks function name `f` against inbuilt library functions/constants of `mpmath` and invoke the corresponding method/value.

- In this case `phi` is actually a `mpmath` constant and `psi` is a method(probably). Please, see [http://mpmath.org/doc/current/functions/constants.html#golden-ratio-phi](url)

**Comment by moorepants** [comment]
That sounds like bad behavior then. I would expect this to operate consistently for all function names. It is odd to expect the user to know that there are some combinations of characters for function names that cause evaluation of arbitrary functions.

**Comment by adityausathe** [comment]
Do you want to get rid of that `mpmath` check? But, we don't why it was there in the first place. I can work on this one. Since, I'm newbie over here, it'll help me to get started.

**Comment by moorepants** [comment]
@asmeurer @smichr Would you all call tis expected behavior or is this unintended?

**Comment by faze-geek** [comment]
`UPDATE`: This has been fixed on master ( I cannot recognize if someone fixed it in a PR or is it a improvement in `mpmath` but anyways gets the job done. )
```
>>> from sympy import *
>>> t = Symbol('t')
>>> phi, psi = symbols('phi, psi', cls=Function)
>>> phi(t)
phi(t)
>>> psi(t)
psi(t)
>>> phi(0)
phi(0)
>>> psi(0)
psi(0)
>>> phi(0.0)
phi(0.0)
>>> psi(0.0)
psi(0.0)
```
Hence this issue can be closed .

**Comment by moorepants** [comment]
Confirmed that this is fixed (checked with sympy 1.10.1).
